# Importing laibraries

In [2]:
import pandas as pd
import numpy as np
from scipy import stats
import statsmodels as sm

# Importing Dataset

In [3]:
data= pd.read_csv('C:/Users/HP/Downloads/Assignment/BuyerRatio.csv')

In [4]:
data

Observed Values  East  West  North  South
0           Males    50   142    131     70
1         Females   435  1523   1356    750

In [4]:
data.columns

Index(['Observed Values', 'East', 'West', 'North', 'South'], dtype='object')

In [10]:
data1=data.drop('Observed Values',axis=1)

In [11]:
data1

East  West  North  South
0    50   142    131     70
1   435  1523   1356    750

In [12]:
data1.values

array([[  50,  142,  131,   70],
       [ 435, 1523, 1356,  750]], dtype=int64)

In [14]:
stats.chi2_contingency(data1)

(1.595945538661058,
 0.6603094907091882,
 3,
 array([[  42.76531299,  146.81287862,  131.11756787,   72.30424052],
        [ 442.23468701, 1518.18712138, 1355.88243213,  747.69575948]]))

#x=male,female  y=sales

#since both the x and y are discrete in 2+categories, we will go for chi-squared test

#Let H0= all proportiona are equal and Ha=Not all proportions are same.

In [15]:
chi2=stats.chi2_contingency(data1)

In [16]:
chi2

(1.595945538661058,
 0.6603094907091882,
 3,
 array([[  42.76531299,  146.81287862,  131.11756787,   72.30424052],
        [ 442.23468701, 1518.18712138, 1355.88243213,  747.69575948]]))

In [18]:
chi2_pvalue=chi2[1]
print('p-value is '+str(chi2_pvalue))


p-value is 0.6603094907091882


#since, P-value=0.66>0.05-->P high Null fly. therefore, all proportions are equal.
